runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -0,0 +1,422 @@
+    #[sdk_error(code = 2)]
+    InvalidDenomination,
+    #[error("withdraw: insufficient runtime balance")]
+    #[sdk_error(code = 3)]
+    InsufficientWithdrawBalance,
+    #[error("core error")]
+    #[sdk_error(code = 4)]
+    CoreError(#[from] modules::core::Error),
+}
+
+/// Events emitted by the consensus module (none so far).
+#[derive(Debug, Serialize, Deserialize, oasis_runtime_sdk_macros::Event)]
+#[serde(untagged)]
+pub enum Event {}
+
+/// Parameters for the consensus module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters {
+    pub consensus_denomination: token::Denomination,
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        Self {
+            // TODO: get this from consensus? (token symbol?).
+            consensus_denomination: token::Denomination::from_str("TEST").expect("to work"),
+        }
+    }
+}
+
+impl module::Parameters for Parameters {
+    type Error = ();
+}
+
+/// Genesis state for the consensus module.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Genesis {
+    #[serde(rename = "parameters")]
+    pub parameters: Parameters,
+}
+
+/// Interface that can be called from other modules.
+pub trait API {
+    /// Deposit an amount into the runtime account.
+    fn deposit<C: Context>(
+        ctx: &mut C,
+        from: Address,
+        amount: token::BaseUnits,
+    ) -> Result<(), Error>;
+
+    /// Withdraw an amount out from the runtime account.
+    fn withdraw<C: Context>(
+        ctx: &mut C,
+        to: Address,
+        amount: token::BaseUnits,
+    ) -> Result<(), Error>;
+
+    // TODO:
+    //  - Add/reclaim deposited escrow.
+    //      - need a way to get escrow events in runtime: https://github.com/oasisprotocol/oasis-core/issues/3862
+}
+
+/// Consensus accounts module.
+///
+/// This module allows consensus transfers in and out of the runtime account,
+/// while keeping track of amount deposited per account.
+pub struct Module<Accounts: modules::accounts::API, Consensus: modules::consensus::API> {
+    _accounts: std::marker::PhantomData<Accounts>,
+    _consensus: std::marker::PhantomData<Consensus>,
+}
+
+const CONSENSUS_TRANSFER_HANDLER: &str = "consensus.TransferFromRuntime";
+const CONSENSUS_WITHDRAW_HANDLER: &str = "consensus.WithdrawIntoRuntime";
+
+impl<Accounts: modules::accounts::API, Consensus: modules::consensus::API> API
+    for Module<Accounts, Consensus>
+{
+    fn deposit<C: Context>(
+        ctx: &mut C,
+        from: Address,
+        amount: token::BaseUnits,
+    ) -> Result<(), Error> {
+        if ctx.is_check_only() {
+            return Ok(());
+        }
+
+        // Ensure correct denomination.
+        let params = Self::params(ctx.runtime_state());
+        let denomination = params.consensus_denomination;
+        if amount.denomination() != &denomination {
+            return Err(Error::InvalidDenomination);
+        }
```

Comment:
Maybe we could factor this out as something like `Self::ensure_consensus_denomination(amount.denomination())?`.

Reply:
Yeah agreed, Done, also moved the denoomination part into the `consesus` module.